include/swift/Basic/LangOptions.h

```diff
@@ -440,6 +440,9 @@ namespace swift {
     /// The name of the package this module belongs to.
     std::string PackageName;
 
+    /// Allow importing a non-package interface from the same package.
+    bool PreferInterfaceImportFromSamePackage = false;
```

Comment:
Importing a non-package interface is already preferred, hence client needs to opt in with EnablePackageInterfaceLoad.

Reply:
Naming is hard. The comment is what this option is really about.

Reply:
This entire patch is about not allowing import non-package interface if it is in the same package because that is not the correct dependency.

Reply:
Can you rename it to reflect the comment to be more clear, e.g. AllowNonInterfaceImportFromSamePackage?